pkgs/applications/graphics/emulsion/default.nix

```
@@ -0,0 +1,85 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, rustPlatform
+, installShellFiles
+, makeWrapper
+, pkg-config
+, python3
+, libGL
+, libX11
+, libXcursor
+, libXi
+, libXrandr
+, libXxf86vm
+, libxcb
+, libxkbcommon
+, wayland
+, xdg_utils
+, AppKit
+, CoreGraphics
+, CoreServices
+, Foundation
+, OpenGL
+}:
+let
+  rpathLibs = [
+    libGL
+    libX11
+    libXcursor
+    libXi
+    libXrandr
+    libXxf86vm
+    libxcb
+  ] ++ lib.optionals stdenv.isLinux [
+    libxkbcommon
+    wayland
+  ];
+in
+rustPlatform.buildRustPackage rec {
+  pname = "emulsion";
+  version = "7.2";
+
+  src = fetchFromGitHub {
+    owner = "ArturKovacs";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1king04p5j4gsvprrfppwaxa5jn4ga4nc0v63wl6fvq2ngwnkg4g";
+  };
+
+  cargoSha256 = "19vb2q5w3063l7349p1b8q40n97a4hxdvc7h7qsx1iiwp75knbd0";
+
+  nativeBuildInputs = [
+    installShellFiles
+    makeWrapper
+    pkg-config
+    python3
+  ];
+
+  buildInputs = rpathLibs ++ lib.optionals stdenv.isDarwin [
+    AppKit
+    CoreGraphics
+    CoreServices
+    Foundation
+    OpenGL
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    install -D $releaseDir/emulsion $out/bin/emulsion
+  '' + lib.optionals (stdenv.isLinux) ''
```

```suggestion
  '' + lib.optional (stdenv.isLinux) ''
```

Try this and tell if something happened.

---

``
error: cannot coerce a list to a string, at /home/bart/source/nixpkgs/pkgs/applications/graphics/emulsion/default.nix:67:18
``

---

Why is optionals failing?

---

I'm not following you.
``optionals`` is working for me an ``optional`` gives me the above error.
ofborg seems to get that error *with* the s.  I'm confused.

---

I am trying to make sense of this:

```
while evaluating the attribute 'installPhase' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-0/pkgs/applications/graphics/emulsion/default.nix:67:3:
cannot coerce a list to a string, at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-0/pkgs/applications/graphics/emulsion/default.nix:67:18
```

---

Yeah, me too!
I get the error when I remove the ``s``, an ofborg gets it with the code in it's current state...

---

```suggestion
  '' + lib.optionalString stdenv.isLinux ''
```

---

Done.